x-pack/plugins/ml/public/application/notifications/components/notifications_list.tsx

```diff
@@ -0,0 +1,320 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC, useCallback, useMemo, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import {
+  EuiBadge,
+  EuiInMemoryTable,
+  EuiSearchBar,
+  EuiSpacer,
+  IconColor,
+  Query,
+  SearchFilterConfig,
+} from '@elastic/eui';
+import { EuiBasicTableColumn } from '@elastic/eui/src/components/basic_table/basic_table';
+import { FIELD_FORMAT_IDS } from '@kbn/field-formats-plugin/common';
+import useDebounce from 'react-use/lib/useDebounce';
+import { useTimeRangeUpdates } from '../../contexts/kibana/use_timefilter';
+import { useToastNotificationService } from '../../services/toast_notification_service';
+import { useFieldFormatter } from '../../contexts/kibana/use_field_formatter';
+import { useRefresh } from '../../routing/use_refresh';
+import { useTableSettings } from '../../data_frame_analytics/pages/analytics_management/components/analytics_list/use_table_settings';
+import { MESSAGE_LEVEL, MessageLevel } from '../../../../common/constants/message_levels';
+import { ListingPageUrlState } from '../../../../common/types/common';
+import { usePageUrlState } from '../../util/url_state';
+import { ML_PAGES } from '../../../../common/constants/locator';
+import type { NotificationItem } from '../../../../common/types/notifications';
+import { useMlKibana } from '../../contexts/kibana';
+
+const levelBadgeMap: Record<MessageLevel, IconColor> = {
+  error: 'danger',
+  info: 'default',
+  success: 'subdued',
+  warning: 'warning',
+};
+
+export const getDefaultNotificationsListState = (): ListingPageUrlState => ({
+  pageIndex: 0,
+  pageSize: 25,
+  sortField: 'timestamp',
+  sortDirection: 'desc',
+});
+
+export const NotificationsList: FC = () => {
+  const {
+    services: {
+      mlServices: { mlApiServices },
+    },
+  } = useMlKibana();
+  const { displayErrorToast } = useToastNotificationService();
+
+  const timeRange = useTimeRangeUpdates();
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [items, setItems] = useState<NotificationItem[]>([]);
+  const [totalCount, setTotalCount] = useState<number>(0);
+
+  const dateFormatter = useFieldFormatter(FIELD_FORMAT_IDS.DATE);
+
+  const [pageState, updatePageState] = usePageUrlState(
+    ML_PAGES.NOTIFICATIONS,
+    getDefaultNotificationsListState()
+  );
+
+  const { onTableChange, pagination, sorting } = useTableSettings<NotificationItem>(
+    totalCount,
+    pageState,
+    updatePageState
+  );
+
+  const refresh = useRefresh();
+
+  const searchQueryText = pageState.queryText;
+
+  const queryInstance = useMemo<Query | undefined>(() => {
+    try {
+      return EuiSearchBar.Query.parse(searchQueryText ?? '');
+    } catch (error) {
+      displayErrorToast(
+        error,
+        i18n.translate('xpack.ml.notifications.invalidQueryError', {
+          defaultMessage: 'Invalid query',
```

Comment:
According to the [Inclusivity guidelines](https://docs.elastic.dev/content-design/inclusivity#avoid-violent-offensive-ableist-terminology) the word "invalid" is not preferable. The guideline suggests using "not valid" or "incorrect" instead.

```suggestion
          defaultMessage: 'Query is not valid',
```

Reply:
Fixed in https://github.com/elastic/kibana/pull/140613/commits/b1a77b9893c21f314acdaa23cd310e457bc62fd4